packages/fx-core/src/plugins/resource/frontend/dotnet/resources/templateInfo.ts

```diff
@@ -0,0 +1,51 @@
+import { PluginContext } from "@microsoft/teamsfx-api";
+import { Capability } from "./../constants";
+import { templatesVersion } from "../../../../../common/template-utils/templates";
+
+export type TemplateVariable = { [key: string]: string };
+
+export class TemplateInfo {
+  group: string;
+  language: string;
+  scenario: string;
+  version: string;
+  variables: TemplateVariable;
+
+  constructor(ctx: PluginContext, group: string, templateVariable: TemplateVariable) {
+    this.group = group;
+    this.version = TemplateInfo.version;
+    this.language = TemplateInfo.DonetLanguage;
+    this.variables = templateVariable;
+    this.scenario = TemplateInfo.DefaultScenario;
+  }
+
+  static readonly SupportedCapabilities = [Capability.tab, Capability.bot];
+  static readonly DonetLanguage = "csharp";
+  static readonly DefaultScenario = "default";
+  static readonly BaseGroup = "blazor-base";
+  static readonly version = templatesVersion;
+}
+
+export function generateTemplateInfos(selectedCapabilities: string[], ctx: PluginContext) {
+  const projectName = ctx.projectSettings!.appName;
+  const templateVariable: TemplateVariable = { BlazorAppServer: projectName };
+  const templateInfoList: TemplateInfo[] = [];
+
+  selectedCapabilities.forEach((capability) => {
+    if (TemplateInfo.SupportedCapabilities.includes(capability)) {
+      templateInfoList.push(new TemplateInfo(ctx, capability.toLowerCase(), templateVariable));
+    }
+  });
+
+  // Generate templateInfo for base scenrio.
+  if (selectedCapabilities.includes(Capability.tab)) {
+    templateVariable.IS_TAB = "true";
+  }
+  if (selectedCapabilities.includes(Capability.bot)) {
```

Comment:
For this method, it could be better to use map instead of if condition to make the code be more generic and extensible for future requirement (e.g. messaging extension support).
```
const supportedCapabilities = new Map<string, string>([
  ["Tab", "IS_TAB"],
  ["Bot", "IS_BOT"],
]);
...

  const variables: string[] = [];

  selectedCapabilities.forEach((capability) => {
    if (supportedCapabilities.has(capability)) {
      templateInfoList.push(new TemplateInfo(ctx, capability.toLowerCase(), templateVariable));
      variables.push(supportedCapabilities.get(capability));
    }
  });
  // add variables to template variable
  variables.forEach(v => templateVariable[v] = "true");
  templateInfoList.push(new TemplateInfo(ctx, TemplateInfo.BaseGroup, templateVariable);

  return templateInfoList;
```

Reply:
In this way, there is not if condition about tab/bot capability so that it is more generic as framework code. If we need to support messaging extension in the future. The only code change is to add key-value pair to supportedCapabilities.
```
const supportedCapabilities = new Map<string, string>([
  ["Tab", "IS_TAB"],
  ["Bot", "IS_BOT"],
  ["MessagingExtension", "IS_MSG"],  <= only add one code here!
]);
```